iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/RatisMetricsManager.java

```diff
@@ -26,12 +26,50 @@
 import org.apache.iotdb.metrics.utils.MetricLevel;
 
 import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
 
 public class RatisMetricsManager {
+
+  static class TimeKeeper implements AutoCloseable {
+    private final long startMoment;
+    private final TConsensusGroupType groupType;
+    private final BiConsumer<Long, TConsensusGroupType> reporter;
+
+    private TimeKeeper(
+        BiConsumer<Long, TConsensusGroupType> reporter, TConsensusGroupType groupType) {
```

Comment:
Can we find a way to avoid the cost of boxing or unboxing long？

Reply:
primitive types are not allowed for Functional interfaces. Besides, the boxing & unboxing costs are trivial in metric reporting operation.